src/segment-loader.js

```diff
@@ -717,6 +717,43 @@ export default class SegmentLoader extends videojs.EventTarget {
     };
   }
 
+  abortRequestEarly_(stats) {
+    const firstByteReceived = Date.now() - stats.firstByteReceived;
+    // Since bandwidth values reported by progress events stabilize as time goes on,
+    // we adjust bandwidth by a conservative ammount, increasing as more time has passed
+    // maxing at 0.8 to prevent aborting a request we actually do have enough bandwidth
+    // for.
+    const playlistBandwidthAdjustment = Math.min(0.8, firstByteReceived / 5000);
```

Comment:
5 seconds is arbitrary. I like the idea of using target duration or the segments specified duration. Maybe half the segment duration? If we've already spent half the duration requesting the segment and bandwidth is still low, it might be a good time to switch. Another idea is what if we incorporate forward buffer so that the closer we get to the end of the buffer, the higher the adjustment ramps up?